### DISCUSSION TITLE
Occasional delay in activating smart plug

### DISCUSSION THREAD

**Discussion by peteS-UK** [make-all/tuya-local#1729] [General]
Some of my smart plugs get an occasional delay in switching their state.  For example, I have a zigbee switch which turns on a plug.  In the log book, I see the integration triggered, but sometimes it might be 20 seconds before the plug switches state and that change shows up in the log book. Most of the time, it's pretty instant.  If I switch it manually using the tuya app, it always seems instant, so I don't think it's network (and that switch is only about 10 feet from an acccess point).  I used to use localtuya, and that was also pretty much instant, but much less reliable with devices going unavailable regularly.  I don't have anything set to Auto in the configs.  Anything I can look at, or is it just the way it is?
Thanks

**Comment by make-all** [comment]
I think it may be a problem in HA 2024.3.x

I am also noticing such delays with my Zigbee (ZHA) and Broadlink IR devices, which are non-Tuya through their own integrations.

**Comment by peteS-UK** [comment]
Ah - that would make more sense.  It's really odd that it just seems to stall - no error, but just sits there for a while....
